nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractComponentNode.java

```diff
@@ -784,13 +784,14 @@ public synchronized void reloadAdditionalResourcesIfNecessary() {
 
         final boolean dynamicallyModifiesClasspath = descriptors.stream()
                 .anyMatch(PropertyDescriptor::isDynamicClasspathModifier);
+        final String isolationKey = determineClasloaderIsolationKey();
 
-        if (dynamicallyModifiesClasspath) {
+        if (dynamicallyModifiesClasspath || isolationKey != null) {
             final Set<URL> additionalUrls = this.getAdditionalClasspathResources(descriptors, this::getEffectivePropertyValueWithDefault);
 
-            final String newFingerprint = ClassLoaderUtils.generateAdditionalUrlsFingerprint(additionalUrls, determineClasloaderIsolationKey());
-            if (!StringUtils.equals(additionalResourcesFingerprint, newFingerprint)) {
-                setAdditionalResourcesFingerprint(newFingerprint);
+            final String newFingerprint = ClassLoaderUtils.generateAdditionalUrlsFingerprint(additionalUrls, isolationKey);
+            if (!StringUtils.equals(componentFingerprint, newFingerprint)) {
+                setComponentFingerprint(newFingerprint);
```

Comment:
I suggest not renaming the variable/method from `additionalResourcesFingerprint` to `componentFingerprint` but leaving it as it is now.
To be fully consistent, the interface methods (`isReloadAdditionalResourcesNecessary()`, `reloadAdditionalResourcesIfNecessary()`) should be renamed too. However, I don't think it is worth changing these API methods.
Also, the isolation key was added in the "additional resources fingerprint" earlier so it was already a bit inconsistent name. It will be no better or worse due to the change in this PR.